lib/executor/constant_arrival_rate.go

```diff
@@ -75,26 +76,31 @@ func NewConstantArrivalRateConfig(name string) ConstantArrivalRateConfig {
 var _ lib.ExecutorConfig = &ConstantArrivalRateConfig{}
 
 // GetPreAllocatedVUs is just a helper method that returns the scaled pre-allocated VUs.
-func (carc ConstantArrivalRateConfig) GetPreAllocatedVUs(es *lib.ExecutionSegment) int64 {
-	return es.Scale(carc.PreAllocatedVUs.Int64)
+func (carc ConstantArrivalRateConfig) GetPreAllocatedVUs(et *lib.ExecutionTuple) int64 {
+	return et.ScaleInt64(carc.PreAllocatedVUs.Int64)
 }
 
 // GetMaxVUs is just a helper method that returns the scaled max VUs.
-func (carc ConstantArrivalRateConfig) GetMaxVUs(es *lib.ExecutionSegment) int64 {
-	return es.Scale(carc.MaxVUs.Int64)
+func (carc ConstantArrivalRateConfig) GetMaxVUs(et *lib.ExecutionTuple) int64 {
+	return et.ScaleInt64(carc.MaxVUs.Int64)
```

Comment:
These probably can be pre-calculated in the constructor, given how many times they are being used below? Something for the follow-up optimization pass perhas?

Reply:
I would prefer if we can merge this before I need to rebase it again :D